src/main/java/com/hubspot/jinjava/util/DeferredValueUtils.java

```diff
@@ -0,0 +1,208 @@
+package com.hubspot.jinjava.util;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.hubspot.jinjava.interpret.Context;
+import com.hubspot.jinjava.interpret.DeferredValue;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.tree.ExpressionNode;
+import com.hubspot.jinjava.tree.Node;
+import com.hubspot.jinjava.tree.TagNode;
+import com.hubspot.jinjava.tree.TextNode;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class DeferredValueUtils {
+  private static final String TEMPLATE_TAG_REGEX = "(\\w+(?:\\.\\w+)*)";
+  private static final Pattern TEMPLATE_TAG_PATTERN = Pattern.compile(TEMPLATE_TAG_REGEX);
+
+  private static final Pattern SET_TAG_PATTERN = Pattern.compile(
+    "set " + TEMPLATE_TAG_REGEX
```

Comment:
Could you use the name defined here? https://github.com/HubSpot/jinjava/blob/5cb246370d8f2bdb304fdcfa22cfa57dfab2c14f/src/main/java/com/hubspot/jinjava/lib/tag/SetTag.java#L65